binding/nodejs/index.js

```diff
@@ -0,0 +1,218 @@
+const MODEL_PATH_DEFAULT = "lib/common/leopard_params.pv";
+
+const VALID_AUDIO_EXTENSIONS = [
+  ".flac",
+  ".mp3",
+  ".ogg",
+  ".opus",
+  ".vorbis",
+  ".wav",
+  ".webm"
+];
+
+/**
+ * Wraps the Leopard engine.
+ *
+ * Performs the calls to the Leopard node library. Does some basic parameter validation to prevent
+ * errors occurring in the library layer. Provides clearer error messages in native JavaScript.
+ */
+class Leopard {
+  /**
+   * Creates an instance of Leopard.
+   * @param {string} accessKey AccessKey obtained from Picovoice Console (https://console.picovoice.ai/).
+   * @param {string} manualModelPath the path to the Leopard model (.pv extension)
+   * @param {string} manualLibraryPath the path to the Leopard dynamic library (platform-dependent extension)
+   */
+  constructor(accessKey, manualModelPath, manualLibraryPath) {
+    if (
+      accessKey === null ||
+      accessKey === undefined ||
+      accessKey.length === 0
+    ) {
+      throw new PvArgumentError(`No AccessKey provided to Leopard`);
+    }
+
+    let modelPath = manualModelPath;
+    if (modelPath === undefined || modelPath === null) {
+      modelPath = path.resolve(__dirname, MODEL_PATH_DEFAULT);
+    }
+
+    let libraryPath = manualLibraryPath;
+    if (libraryPath === undefined || modelPath === null) {
+      libraryPath = getSystemLibraryPath();
+    }
+
+    if (!fs.existsSync(libraryPath)) {
+      throw new PvArgumentError(
+        `File not found at 'libraryPath': ${libraryPath}`
+      );
+    }
+
+    if (!fs.existsSync(modelPath)) {
+      throw new PvArgumentError(`File not found at 'modelPath': ${modelPath}`);
+    }
+
+    const pvLeopard = require(libraryPath);
+
+    let leopardHandleAndStatus = null;
+    try {
+      leopardHandleAndStatus = pvLeopard.init(accessKey, modelPath);
+    } catch (err) {
+      pvStatusToException(PV_STATUS_T[err.code], err);
+    }
+
+    const status = leopardHandleAndStatus.status;
+    if (status !== PV_STATUS_T.SUCCESS) {
+      pvStatusToException(status, "Leopard failed to initialize");
+    }
+
+    this._handle = leopardHandleAndStatus.handle;
+    this._pvLeopard = pvLeopard;
+    this._sampleRate = pvLeopard.sample_rate();
+    this._version = pvLeopard.version();
+  }
+
+  /**
+   * @returns the audio sampling rate accepted by the process function
+   * @see {@link process}
+   */
+  get sampleRate() {
+    return this._sampleRate;
+  }
+
+  /**
+   * @returns the version of the Leopard engine
+   */
+  get version() {
+    return this._version;
+  }
+
+  /**
+   * Processes a given audio data and returns its transcription.
+   *
+   * @param {Array} pcm Audio data. The audio needs to have a sample rate equal to `.sampleRate` and be 16-bit linearly-encoded.
+   * This function operates on single-channel audio. If you wish to process data in a different
+   * sample rate or format consider using `.processFile`.
+   * @returns {string} Inferred transcription.
+   */
+  process(pcm) {
+    if (
```

Comment:
can this happen?

Reply:
if the object is released and then `process` is called, `this._handle` is zero